addons/account/models/account_move.py

```diff
@@ -4766,9 +4745,36 @@ def _message_post_after_hook(self, new_message, message_values):
                 invoice.attachment_ids |= attachments
                 invoice.message_ids = [Command.set(sub_new_message.id)]
                 super(AccountMove, invoice)._message_post_after_hook(sub_new_message, sub_message_values)
-
         return res
 
+    def _check_and_decode_attachment(self, attachments):
+        if not attachments or self.env.context.get('no_new_invoice') or not self.is_invoice(include_receipts=True):
+            return False
+        attachments_in_invoices = self.env['ir.attachment']
+        odoobot = self.env.ref('base.partner_root')
+        if self.state != 'draft':
+            self.message_post(body=_('The invoice is not a draft, it was not updated from the attachment.'),
+                              message_type='comment',
+                              subtype_xmlid='mail.mt_note',
+                              author_id=odoobot.id)
+            return False
+
+        # As we are coming from the mail, we assume that ONE of the attachments
+        # will enhance the invoice thanks to EDI / OCR / .. capabilities
+        has_existing_lines = bool(self.invoice_line_ids)
+        results = self._extend_with_attachments(attachments, new=bool(self._context.get('from_alias')))
+        if has_existing_lines and not results:
+            self.message_post(body=_('The invoice already contains lines, it was not updated from the attachment.'),
+                              message_type='comment',
+                              subtype_xmlid='mail.mt_note',
+                              author_id=odoobot.id)
+            return False
+        for attachment in results:
+            attachments_in_invoices += attachment
```

Comment:
moved the definition of the variable before the loop, exclusively for your pretty eyes since I personally prefer for the variables to be declared beforehand :P 
Adding` .keys() `raises a checkstyle error though, so your pretty eyes won't suffice.